src/items.rs

```diff
@@ -1338,7 +1313,7 @@ pub(crate) fn format_struct_struct(
             context,
             g,
             context.config.brace_style(),
-            set_brace_pos(&context, &struct_parts, &fields, body_lo),
+            set_struct_brace_pos(&context, &fields, mk_sp(body_lo, span.hi() - BytePos(1))),
```

Comment:
Instead of creating the `inner_span` inline with `mk_sp(body_lo, span.hi() - BytePos(1))` can we just hoist it up from where it's [originally defined](https://github.com/rust-lang/rustfmt/blob/ef91154250977b3b5d05448dafbca524a1168b47/src/items.rs#L1329). 

```diff
+ let inner_span = mk_sp(body_lo, span.hi() - BytePos(1));
  let generics_str = match struct_parts.generics {
         Some(g) => format_generics(
             context,
             g,
             context.config.brace_style(),
-            set_brace_pos(&context, &struct_parts, &fields, body_lo),
+            set_struct_brace_pos(&context, &fields, inner_span)),
             ..
         )
         ..
 };
 ...
     if fields.is_empty() {
-        let inner_span = mk_sp(body_lo, span.hi() - BytePos(1));
         format_empty_struct_or_tuple(context, inner_span, offset, &mut result, "", "}");
         return Some(result);
     }
```

Reply:
@ytmimi Hey! Hope you're doing awesome!

I was about to ask you a question about this point hehe (hositing the `inner_span` on the outermost scope of the function). I'll add the changes in a few minutes 👍.